accountability.py

```diff
@@ -353,16 +407,69 @@ def work():
     else:
         pair_id = 0
 
+    # Manually setting work step
+    if job == JOB_OBS_VAL:
+        step = STEP_WORK_1_VAL
+    else:
+        step = STEP_WORK_2_VAL
+
     color, page, reason = get_page_values(job, step, condition)
     room_name = '{}|{}'.format(obs, mod) if condition == CONDITION_EXP_VAL else ''
     print(room_name)
 
 	# Getting all image URLs
     all_imgs = query_db('select img_path from images')
-    subset = get_array_subset(all_imgs, NUM_IMAGES)
+
+    # Getting list of images this user has already observed
+    # Saving, retrieving, and/or generating random subset for this pair
+    chosen_imgs = query_db('select img_id from chosen_imgs where pair_id=?', [pair_id])
+    if chosen_imgs is None or len(chosen_imgs) == 0:
+        curr_mod = query_db('select mod_id from pairs where id=?', [pair_id], one=True)
+        cannot_contain = []
+        if curr_mod is not None:
+            last_pair = query_db('select id from pairs where obs_id=?', [curr_mod[0]], one=True)
+            if last_pair is not None:
+                cannot_contain_ids = query_db('select img_id from moderations where pair_id=?', [last_pair[0]])
+                for id in cannot_contain_ids:
```

Comment:
There might be a better way here. Can you do something like `query_db('select img_path from images where img_id in ?', cannot_contain_ids)`?

Might be worth looking into this, it'll mean you don't have to iterate over every single id.